src/bin/bat/config.rs

```diff
@@ -15,6 +16,46 @@ pub fn config_file() -> PathBuf {
         .unwrap_or_else(|| PROJECT_DIRS.config_dir().join("config"))
 }
 
+pub fn generate_config_file() {
+    let config_file = config_file();
+    if config_file.exists() {
+        println!("A config file already exists at: {}", config_file.to_string_lossy());
+
+        print!("Overwrite? (y/n): ");
+        let _ = io::stdout().flush();
+        let mut decision = String::new();
+        io::stdin().read_line(&mut decision).expect("Failed to read input");
+
+        if !decision.trim().eq_ignore_ascii_case("Y") {
+            return;
+        }
+    } else {
+        let config_dir = config_file.parent().unwrap();
+        if !config_dir.exists() {
```

Comment:
Doing a check like this before creating a directory or file is generally discouraged because it can lead to race conditions. Even if that is unlikely to be a problem here, the alternative is much simpler: simply create the directory and ignore the fact that it could already exist.

Now `fs::create_dir` would return an error in case the path already exists, but you can (and should) use `fs::create_dir_all` anyway. `create_dir` would fail if the parent directory would not exist (`~/.config` on Unix systems).